ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ProfilesToggleGroup.tsx

```diff
@@ -1,29 +1,115 @@
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { useParams } from 'react-router-dom';
-import { ToggleGroup, ToggleGroupItem } from '@patternfly/react-core';
+import { Tab, Tabs, ToggleGroup, ToggleGroupItem } from '@patternfly/react-core';
 
-import { ComplianceProfileScanStats } from 'services/ComplianceResultsStatsService';
+import { ComplianceProfileSummary } from 'services/ComplianceCommon';
+
+const NON_STANDARD_TAB = 'Other';
+
+// Extract unique standards from profiles and add 'Other' standard if there are profiles with no standards
+function getUniqueStandards(profiles: ComplianceProfileSummary[]): string[] {
+    const standards = new Set(
+        profiles.flatMap((profile) => profile.standards.map((standard) => standard.shortName))
+    );
+    if (profiles.some((profile) => profile.standards.length === 0)) {
+        standards.add(NON_STANDARD_TAB);
+    }
+    return Array.from(standards);
+}
+
+function getInitialStandard(
+    profiles: ComplianceProfileSummary[],
+    profileNameParam: string
+): string {
+    const profile = profiles.find((profile) => profile.name === profileNameParam);
+    if (profile && profile.standards.length > 0) {
+        return profile.standards[0].shortName;
+    }
+    return NON_STANDARD_TAB;
+}
+
+function isStandardInProfile(standardShortName: string, profile: ComplianceProfileSummary) {
+    return (
+        profile.standards.some((standard) => standard.shortName === standardShortName) ||
+        (standardShortName === 'Other' && profile.standards.length === 0)
+    );
+}
 
 type ProfilesToggleGroupProps = {
-    profiles: ComplianceProfileScanStats[];
+    profiles: ComplianceProfileSummary[];
     handleToggleChange: (selectedProfile: string) => void;
 };
 
 function ProfilesToggleGroup({ profiles, handleToggleChange }: ProfilesToggleGroupProps) {
     const { profileName: profileNameParam } = useParams();
 
+    const uniqueStandards = useMemo(() => getUniqueStandards(profiles), [profiles]);
+    const initialStandard = useMemo(
+        () => getInitialStandard(profiles, profileNameParam),
+        [profileNameParam, profiles]
+    );
+
+    const [selectedStandard, setSelectedStandard] = useState(initialStandard);
+
+    useEffect(() => {
+        // Sets the selected standard based on the profile name in the URL.
+        // Currently picks the first standard found since no profile should have multiple standards, however
+        // if this changes in the future, we'll want to find all matches and only update selectedStandard if the
+        // current selectedStandard doesn't exist in the match
+        if (profileNameParam) {
+            const standardShortName =
+                profiles.find((profile) => profile.name === profileNameParam)?.standards[0]
+                    ?.shortName || NON_STANDARD_TAB;
+            setSelectedStandard(standardShortName);
+        }
+    }, [profileNameParam, profiles]);
```

Comment:
Possible follow up to make cause-and-effect even clearer:
* replace `const { profileName: profileNameParam } = useParams();` in component
* with `profileName` prop because the callers get `profileName` from page address params

Reply:
Agreed, will update. That would be more consistent with what I've done in other parts as well.